src/languages/swift.js

```diff
@@ -283,6 +288,143 @@ export default function(hljs) {
   };
   TYPE.contains.push(GENERIC_ARGUMENTS);
 
+  // https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID552
+  // Prevents element names from being highlighted as keywords.
+  const TUPLE_ELEMENT_NAME = {
+    begin: lookahead(concat(Swift.identifier, /\s*:/)),
+    end: lookahead(/:/),
+    keywords: "_",
+    relevance: 0
+  };
+  // Matches tuples as well as the parameter list of a function type.
+  const TUPLE = {
+    begin: /\(/,
+    end: /\)/,
+    relevance: 0,
+    keywords: KEYWORDS,
+    contains: [
+      'self',
+      TUPLE_ELEMENT_NAME,
+      ...COMMENTS,
+      ...KEYWORD_MODES,
+      ...BUILT_INS,
+      ...OPERATORS,
+      NUMBER,
+      STRING,
+      ...IDENTIFIERS,
+      ...ATTRIBUTES,
+      TYPE
+    ]
+  };
+
+  // https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID362
+  // Matches both the keyword func and the function title.
+  // Grouping these lets us differentiate between the operator function <
+  // and the start of the generic parameter clause (also <).
+  const FUNCTION_TITLE = {
+    begin: lookahead(/\bfunc\b/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /\bfunc\b/
+      },
+      {
+        begin: /\s+/,
+        relevance: 0
+      },
+      {
+        className: 'title',
+        begin: either(QUOTED_IDENTIFIER.begin, Swift.identifier, Swift.operator),
+        // Required to make sure the opening < of the generic parameter clause
+        // isn't parsed as a second title.
+        endsParent: true,
+        relevance: 0
+      }
+    ]
+  };
+  const GENERIC_PARAMETERS = {
+    begin: /\s*</,
+    end: />/,
+    contains: [
+      ...COMMENTS,
+      TYPE
+    ]
+  };
+  const FUNCTION_PARAMETER_NAME = {
+    begin: either(
+      lookahead(concat(Swift.identifier, /\s*:/)),
+      lookahead(concat(Swift.identifier, /\s+/, Swift.identifier, /\s*:/))
+    ),
+    end: lookahead(/:/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /\b_\b/
```

Comment:
Is `_` truly a keyword in Swift?

Reply:
Yes, it's used as a wildcard pattern.
Which reminds me that I should also add support for other patterns in TUPLE, namely `(let name, var otherName)` etc.

Reply:
Why is an explicit rule needed for that?  `let` and `var` are already keywords, no?

Reply:
Yes, correct. I didn't need a rule for it because I was only matching something with a trailing colon.

My brain was confusing this with the rule for function parameters, which allows two identifiers.